src/mono/mono/mini/aot-runtime.c

```diff
@@ -1971,6 +1971,12 @@ load_aot_module (MonoAssemblyLoadContext *alc, MonoAssembly *assembly, gpointer
 	if (mono_aot_mode == MONO_AOT_MODE_NONE)
 		return;
 
+#ifdef __EMSCRIPTEN__
+	// This indicates that we were not built for AOT, so there's no need to probe for AOT modules.
+	if (mono_aot_mode == MONO_AOT_MODE_INTERP_ONLY)
+		return;
```

Comment:
I don't remember which AOT mode it is, but I think for some interpreter modes we still need the AOT image for corlib because it contains some trampolines

Reply:
ah, that's right. But if I remember right, we only need the aot image for corlib?

Reply:
If I remember correctly, I think wasm is ok here.